### PR TITLE
E2e tests

### PR DESCRIPTION
**Pull request by odra**
## Motivation
Keycloak operator e2e tests

## What
PR contains keycloak operator-sdk framework e2e tests (still WIP)

## Why
Add an short answer for: Why it was done? (E.g The feature X was deprecated.)

## How
Add an short answer for: How it was done? (E.g By removing this feature from ... OR By removing just the button but not its implementation ... ) 

## Verification Steps

Change the image name/tag in `deploy/operator.yaml` if using a different image name.

Build the image:
```
operator-sdk build --enable-tests quay.io/integreatly/keycloak-operator:test
```

Push it to a registry:

```
docker push quay.io/integreatly/keycloak-operator:test
```

Change the image name in deploy/test-e2e.yaml if needed as well.

Add roles, role bindings and crds in your namespace:

```
oc apply -f deploy/rbac.yaml -n keycloak-test
oc apply -f deploy/Keycloak_crd.yaml -n keycloak-test
oc apply -f deploy/KeycloakRealm_crd.yaml -n keycloak-test
```

Deploy the test pod:

```
oc apply -f deploy/test-e2e.yaml -n keycloak-test
```

You can check the pod(s) logs to see how tests are going. 

## Checklist:

- [ ] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Progress

- [ ] Finished task
- [ ] TODO

### CODE REVIEW

File: pkg/keycloak/phaseHandler.go [hunk 1]
```diff
@@ -53,13 +53,18 @@ func (ph *phaseHandler) Initialise(sso *v1alpha1.Keycloak) (*v1alpha1.Keycloak,
...
 
 func (ph *phaseHandler) Accepted(sso *v1alpha1.Keycloak) (*v1alpha1.Keycloak, error) {
-
+	var err error
 	kc := sso.DeepCopy()
-	if kc.Spec.AdminCredentials != "" {
-		return nil, nil
+	adminPwd := kc.Spec.AdminCredentials
+
+	if adminPwd == "" {
```
Comment (philbrookes) [inline pkg/keycloak/phaseHandler.go]: Seeing that this is done whether or not adminPwd is "", do we need this if statement?

File: pkg/keycloak/phaseHandler.go [hunk 2]
```diff
@@ -53,13 +53,18 @@ func (ph *phaseHandler) Initialise(sso *v1alpha1.Keycloak) (*v1alpha1.Keycloak,
...
+		adminPwd, err = GeneratePassword()
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	adminPwd, err := GeneratePassword()
+	adminPwd, err = GeneratePassword()
```
Comment (philbrookes) [inline pkg/keycloak/phaseHandler.go]: or maybe this is not supposed to be here?

**Comment by philbrookes** [comment]
Could we add some instruction to the readme - or a makefile instruction - on how to run this locally?

File: test/e2e/utils.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+func doDeployment(f *framework.Framework, ctx *framework.TestCtx, keycloakCr *v1alpha1.Keycloak, realmCr *v1alpha1.KeycloakRealm) error {
+
+	err := f.Client.Create(goctx.TODO(), realmCr, cleanupOpts(ctx))
+	if err != nil {
+		return err
+	}
+
+	err = f.Client.Create(goctx.TODO(), keycloakCr, cleanupOpts(ctx))
```
Comment (maleck13) [inline test/e2e/utils.go]: why do we create the realm before the keycloak, does it matter?

Comment (odra) [reply test/e2e/utils.go]: the order doesn't really matter but we need both CRs for the deployment to happen.

**Pull request merged**